src/main/scala/nl/biopet/bioconda/BiocondaUtils.scala

```diff
@@ -0,0 +1,209 @@
+    val matches = versionRegex.findAllIn(yaml).matchData
+    if (matches.isEmpty) {
+      throw new Exception(s"No version found in: ${metaYaml.getPath}")
+    }
+    val version = matches.toList.head.group(1)
+    version
+  }
+
+  def dockerInstalled(log: ManagedLogger, path: Option[String] = None): Unit = {
+    val testCommand = "docker version -f '{{.Client.Version}}'"
+    def test: ProcessBuilder =
+      if (path.isDefined) {
+        Process(Seq("bash", "-c", testCommand),
+                None,
+                "PATH" -> path.getOrElse("$PATH"))
+      } else {
+        Process(Seq("bash", "-c", testCommand), None)
+      }
+    try { test.!!(log) } catch {
+      case e: Exception =>
+        throw new Exception(s"Docker does not run: ${e.getMessage}")
+    }
+  }
+  def circleCiCommand(cwd: File, args: Seq[String], log: ManagedLogger) = {
+    val path = cwd.getPath
+    val command = Seq("docker",
+                      "run",
+                      "--rm",
+                      "-v",
+                      "/var/run/docker.sock:/var/run/docker.sock",
+                      "-v",
+                      s"$path:$path",
+                      "--workdir",
+                      s"$path",
+                      "circleci/picard",
+                      "circleci") ++ args
+    (Process(command, cwd).lineStream(log)).foreach(line => log.info(line))
+    // if (exit != 0) throw new Exception(s"Command ${command.mkString(" ")} failed with exit code: ${exit}.")
+  }
+
+  def pullLatestUtils(log: ManagedLogger) = {
+    Process(Seq("docker", "pull", "bioconda/bioconda-utils-build-env"))
+      .lineStream(log)
+      .foreach(line => log.info(line))
+  }
+
+  def copyDirectory(source: File,
+                    dest: File,
+                    permissions: Boolean = true): Unit = {
+    assert(source.isDirectory, "Source should be a directory")
+    if (dest.exists()) {
+      if (!dest.isDirectory) {
+        throw new IOException(
+          s"Destination ${dest.getAbsolutePath} is a file, not a directory.")
+      }
+    } else {
+      dest.mkdirs()
+    }
+    for (file <- source.listFiles()) {
+      val destination = new File(dest, file.getName)
+      if (file.isDirectory) {
+        destination.mkdir()
+        copyDirectory(file, destination, permissions = permissions)
+      } else {
+        // Simple file copy is used here for maximum control.
+        FileUtils.copyFile(file, destination)
+        if (permissions) {
+          destination.setReadable(file.canRead)
+          destination.setWritable(file.canWrite)
+          destination.setExecutable(file.canExecute)
+        }
+      }
+    }
+  }
+
+  def crawlRecipe(recipe: File): Seq[File] = {
+    val files = recipe.listFiles()
+    val yamls = new ArrayBuffer[File]()
+    for (file <- files) {
+      if (file.isDirectory) {
+        yamls ++= crawlRecipe(file)
+      }
+      if (file.getName == "meta.yaml") {
+        yamls.append(file)
+      }
+    }
+    yamls
```

Comment:
This can be done without a mutable buffer. not sure what you trying to do here because of missing scaladocs but I think this is what you want:
```scala
def crawlRecipe(recipe: File): Seq[File] = {
  recipe.listFiles().flatMap { file =>
    if (file.isDirectory) crawlRecipe(file)
    else Seq(file)
  }
}
```